packages/bpk-component-slider/src/BpkSlider.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import Slider from 'react-slider';
+import { cssModules } from 'bpk-react-utils';
+import React, { Component } from 'react';
+
+import STYLES from './bpk-slider.scss';
+
+const getClassName = cssModules(STYLES);
+
+class BpkSlider extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { value: props.value };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(value) {
+    this.setState({ value });
+    this.props.onChange(value);
+  }
+
+  render() {
+    const {
+      min,
+      max,
+      step,
+      large,
+      valueTemplate,
+      minDistance,
+      className,
+      onAfterChange,
+    } = this.props;
+
+    const sliderClass = [];
+    const handleClass = [];
+    const barClass = [];
+
+    sliderClass.push(getClassName('bpk-slider'));
+    sliderClass.push(getClassName(this.state.value.length
+      ? 'bpk-slider__range'
```

Comment:
these classnames should be modifiers i.e. `bpk-slider--range` & `bpk-slider--simple`. Also you can default to simple and only add one class based on whether it's a range or not.

For reference:
```
.calculator {} // block 
.calculator--alternate {} // block modifier 
.calculator__definitions {} // element 
.calculator__definitions--expanded {} // element modifier
```